provider/provider.go

```diff
@@ -1339,10 +1350,22 @@ func setFieldViaOp(typ string, attrs map[string]interface{}, name string, key st
 	var diags diag.Diagnostics
 
 	valStr := attrValueString(typ, key, val, attrs)
-	appendActionLog(fmt.Sprintf("Setting %s field: '%s'.'%s' :: %+v\n", typ, name, key, val))
-
 	op := fmt.Sprintf("%s.%s = %s", name, key, valStr)
 
+	if typ == "dashboard" {
+		isPrimary := GetNestedValueOrDefault(attrs, ToKeyPath(key+".primary"), false).(bool)
+		if isPrimary {
+			appendActionLog(fmt.Sprintf("Skipping setting %s field %s...\n", typ, key))
+			return nil
+		} else {
+			if key == "groups" || key == "values" {
+				op = fmt.Sprintf("%s.%s = %s", name, key, val)
+			}
+		}
+	}
+
+	appendActionLog(fmt.Sprintf("Setting %s field: '%s'.'%s' :: %+v\n", typ, name, key, val))
```

Comment:
`groups` and `values` are b64json in terraform, `attrValueString` will return them as base64 encoded, but we don't want to send them like that.
> I don't understand why we skip primary here.

For the first terraform apply, two things happen:
1. a dashboard object is created using the `name` and the `dashboard_type` (the primary)
2. the fields are updated using the CLI syntax. 
The problem is that when the fields are updated, `dashboard_type` was also included and I didn't want that because I added a validation in the backend that returns an error if the user tries to update this field.

Reply:
So, for any other TF resource, we do a second update for the primary attribute ? 😕